executor/src/main/java/com/redhat/service/bridge/executor/filters/FilterEvaluatorFactoryFEEL.java

```diff
@@ -22,28 +26,35 @@ public FilterEvaluator build(Set<BaseFilter> filters) {
         return new FilterEvaluatorFEEL(templates);
     }
 
-    protected String getTemplateByFilterType(BaseFilter filter) {
+    protected String getTemplateByFilterType(BaseFilter<?> filter) {
         return String.format(TEMPLATE, getFilterCondition(filter));
     }
 
-    private String getFilterCondition(BaseFilter filter) {
+    private String getFilterCondition(BaseFilter<?> filter) {
         switch (filter.getType()) {
             case StringEquals.FILTER_TYPE_NAME:
-                return String.format("%s = \"%s\"", filter.getKey(), filter.getValueAsString());
+                return String.format("%s = \"%s\"", filter.getKey(), filter.getValue());
             case StringContains.FILTER_TYPE_NAME:
-                return getFilterConditionForListValues("(contains (%s, \"%s\"))", filter);
+                return getFilterConditionForListValues("(contains (%s, %s))", (StringContains) filter);
             case StringBeginsWith.FILTER_TYPE_NAME:
-                return getFilterConditionForListValues("(starts with (%s, \"%s\"))", filter);
+                return getFilterConditionForListValues("(starts with (%s, %s))", (StringBeginsWith) filter);
+            case ValuesIn.FILTER_TYPE_NAME:
+                return getFilterConditionForListValues("%s = %s", (ValuesIn) filter);
             default:
                 throw new IllegalArgumentException("Filter type " + filter.getType() + " is not supported by FEELTemplateFactory.");
         }
     }
 
-    @SuppressWarnings("unchecked")
-    private String getFilterConditionForListValues(String singleFormatTemplate, BaseFilter filter) {
+    private <T> String getFilterConditionForListValues(String singleFormatTemplate, BaseFilter<List<T>> filter) {
         List<String> conditions = new ArrayList<>();
-        for (String value : (List<String>) filter.getValue()) {
-            conditions.add(String.format(singleFormatTemplate, filter.getKey(), value));
+        ObjectMapper objectMapper = ObjectMapperFactory.get();
+        for (T value : filter.getValue()) {
+            try {
+                conditions.add(String.format(singleFormatTemplate, filter.getKey(), objectMapper.writeValueAsString(value)));
+            } catch (JsonProcessingException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
```

Comment:
Log? 
Just before I forget about it, I have a more general consideration: in the future we will have to notify the user about runtime issues like this one. If you agree on this point, may I ask you to create an epic like "runtime exception monitoring/notification for user"? I will be complex feature to design and maybe to implement, but it's something we might have to add in the future 
cc @robpblake @danielezonca

Reply:
In this case, this was an omission on my side, this should throw IllegalArgumentException (and I think we probably need an entry in our error catalog for this runtime, but lets discuss that)

Reply:
We have a ticket about defining error strategy and this is definitely a good use case for that

Reply:
Hey @fjtirado , the problem is that this error is thrown in the `executor` application, so the user does not have any visibility on this application yet. It's a kind of internal service that the user can not see

Reply:
@fjtirado @r00ta @danielezonca I'm planning on covering this error scenario as part of the work for https://issues.redhat.com/browse/MGDOBR-145 this sprint 👍